src/runtime_src/core/tools/xbmgmt2/flash/xspi.cpp

```diff
@@ -752,25 +764,25 @@ int XSPI_Flasher::writeReg(unsigned int RegOffset, unsigned int value)
 
 
 bool XSPI_Flasher::waitTxEmpty() {
-    long long delay = 0;
-    while (delay < 30000000000) {
+    long long _delay = 0;
```

Comment:
As you know, I don't like magic numbers. This loop makes a lot of assumptions, but if one does the calculations (e.g. 30000000000*5us) the total delay here is 3 seconds.  To better help the user it would be better to rewrite the code to reflect that desire.

Similar to the delay() code that was written earlier, create a loop around this code that will expire after 3 seconds and produce the error message when it does.

Reply:
This is a great catch! Thanks for explaining it :)